src/zap.ts

```diff
@@ -35,6 +45,30 @@ export function getZapInDetails({ chainId, token, vault }: ZapInArgs): ZapInDeta
   return { isZapInSupported: false, zapInWith: null };
 }
 
+export function getZapOutDetails({ chainId, token, vault }: ZapOutArgs): ZapOutDetails {
+  if (!token.supported || !vault.metadata?.zapOutWith) {
+    return { isZapOutSupported: false, zapOutWith: null };
+  }
+
+  if (isEthereum(chainId)) {
+    const { zapOutWith } = vault.metadata;
+
+    if (zapOutWith && isValidZap(zapOutWith, token.supported)) {
+      const isZapOutSupported = Boolean(token.supported.zapper) && Boolean(token.supported[zapOutWith]);
```

Comment:
do we actually need to call Boolean on this? Can't we just `!!`?

Reply:
It's just my preference, it's clearer what's doing. Do you have a strong preference for `!!`?

Reply:
No strong preferences, it's just that we normally use `!!` both on sdk and yearn-finance-v3 and just to keep consistency accross the codebase i would suggest we keep using `!!`

<img width="265" alt="Screenshot 2022-05-09 at 11 47 34" src="https://user-images.githubusercontent.com/94012134/167436097-176fc50e-f3e8-408d-ac7d-6d0573c56a4c.png">
<img width="250" alt="Screenshot 2022-05-09 at 11 47 47" src="https://user-images.githubusercontent.com/94012134/167436109-4580a566-bbf7-48c6-891f-04f0492b0753.png">

Reply:
I'm always pro-readability, for me `Boolean(value)` is more readable than the double negation `!!value` -- would be happy to refactor the codebase in my next PR to address that and move everything to `Boolean(value)` as I have some preference for that :)